libnavigation-core/src/test/java/com/mapbox/navigation/core/routeoptions/MapboxRouteOptionsUpdaterParameterizedTest.kt

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.navigation.core.routeoptions
+
+import android.location.Location
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.internal.route.RouteUrl
+import com.mapbox.navigation.base.trip.model.RouteLegProgress
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class MapboxRouteOptionsUpdaterParameterizedTest(
+    private val initWaypointNames: String,
+    private val initWaypointIndices: String,
+    private val remainingWaypoints: Int,
+    private val expectedWaypointNames: String,
+    private val expectedWaypointIndices: String
+) {
+
+    companion object {
+
+        @JvmStatic
+        @Parameterized.Parameters
+        fun data() = listOf(
+            arrayOf("start;finish", "0;6", 6, "start;finish", "0;6"),
+            arrayOf("start;finish", "0;6", 4, "start;finish", "0;4"),
+            arrayOf("start;finish", "0;6", 1, "start;finish", "0;1"),
+            arrayOf("start;mid;finish", "0;2;6", 5, "start;mid;finish", "0;1;5"),
+            arrayOf("start;mid;finish", "0;2;6", 4, "mid;finish", "0;4"),
+            arrayOf("start;mid;finish", "0;2;6", 3, "mid;finish", "0;3"),
+            arrayOf("start;mid1;mid2;finish", "0;2;4;6", 5, "start;mid1;mid2;finish", "0;1;3;5"),
+            arrayOf("start;mid1;mid2;finish", "0;2;4;6", 4, "mid1;mid2;finish", "0;2;4"),
+            arrayOf("start;mid1;mid2;finish", "0;2;4;6", 3, "mid1;mid2;finish", "0;1;3"),
+            arrayOf("start;mid1;mid2;finish", "0;2;4;6", 2, "mid2;finish", "0;2"),
+            arrayOf("start;mid1;mid2;finish", "0;2;4;6", 1, "mid2;finish", "0;1"),
+            arrayOf("start;mid1;mid2;finish", "0;2;3;6", 4, "mid1;mid2;finish", "0;1;4"),
+            arrayOf("start;mid1;mid2;finish", "0;2;3;6", 3, "mid2;finish", "0;3"),
+            arrayOf(
+                "start;mid1;mid2;mid3;mid4;mid5;finish",
+                "0;1;2;3;4;5;6",
+                6,
+                "start;mid1;mid2;mid3;mid4;mid5;finish",
+                "0;1;2;3;4;5;6"
+            ),
+            arrayOf(
+                "start;mid1;mid2;mid3;mid4;mid5;finish",
+                "0;1;2;3;4;5;6",
+                2,
+                "mid4;mid5;finish",
+                "0;1;2"
```

Comment:
Could we extract "magic numbers" into constants?

Reply:
I don't think we need to move it into constants, because it is `parameterized` tests and not magic numbers. This is value of `remainingWaypoints` mentioned in constructor.

Reply:
I see a bunch of repeated strings all over the place 😅